### PR TITLE
fix: Fluent markdown fixes - web

### PR DESCRIPTION
**Pull request by JGibson2019**
#### Details
Addresses the two improper nesting issues in the Parsing section listed [here ](https://microsoft-my.sharepoint.com/:w:/p/madparker/EU5Pr5mI8IFOn_wxQVYRpBABXs1C0tyoCZnWofEBZsZBAw?e=5Lif51)

Updated Collapsible Component to have proper nesting. Additionally, created a custom component to handle situations where we have a h1 and guidance link to ensure proper nesting. These should result in no UI changes for users. 
##### Context

Yes, I left out the one markdown fix for the unified application 

#### Pull request checklist

- [X] Addresses an existing issue: #0000
- [X] Ran `yarn fastpass`
- [X] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [X] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/common/components/header-with-content-link.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (madalynrose) [inline src/common/components/header-with-content-link.tsx:1]: I'm rethinking the name of this again. Could we change it to `HeadingWithContentLink` instead? I think that would clarify that it does not contain a `<header>` element.

File: src/common/components/header-with-content-link.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    deps: ContentLinkDeps;
+    headerClass?: string;
+    headingTitleClassName?: string;
+    headingTitle: string;
+    secondaryText?: string;
+    guidance?: ContentReference;
+}
+
+export const HeaderWithContentLink = NamedFC<HeaderWithContentLinkProps>(
+    'HeaderWithContent Link',
```
Comment (madalynrose) [inline src/common/components/header-with-content-link.tsx:19]: Looks like this extra space is propagating through to the snapshot markdown and looks odd. 
```suggestion
    'HeaderWithContentLink',
```

**Review by madalynrose — CHANGES_REQUESTED** [review]

File: src/common/components/header-with-content-link.tsx (hunk 1 above)
Comment (JGibson2019) [reply src/common/components/header-with-content-link.tsx:1]: Ok

File: src/common/components/header-with-content-link.tsx (hunk 2 above)
Comment (JGibson2019) [reply src/common/components/header-with-content-link.tsx]: Updated this in the last commit so resolving this

**Comment by JGibson2019** [comment]
@madalynrose did the renaming! Lmk if there are other requested changes?

**Pull request merged**